modules/back-end/src/Infrastructure/Redis/RedisReadinessCheck.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Infrastructure.Redis;
+
+public class RedisReadinessCheck : IHealthCheck
+{
+    private readonly IRedisClient _redisClient;
+
+    public RedisReadinessCheck(IRedisClient redisClient)
+    {
+        _redisClient = redisClient;
+    }
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            await _redisClient.GetDatabase().PingAsync();
```

Comment:
We can add a method `Task<TimeSpan> PingAsync();`  to `IRedisClient` interface, then in the `DefaultRedisClient`

```csharp
public Task<TimeSpan> PingAsync() => GetDatabase().PingAsync();
```

For mock

```csharp
_mockedClient.Setup(client => client.PingAsync()).Returns(Task.FromResult(TimeSpan.Zero));

// or
_mockedClient.Setup(client => client.PingAsync()).ThrowsAsync(testRedisError);
```

Reply:
I have added `PingAsync` to the redis client and interface 👍